azurerm/internal/services/network/firewall_resource.go

```diff
@@ -134,11 +149,21 @@ func resourceArmFirewallCreateUpdate(d *schema.ResourceData, meta interface{}) e
 		Location: &location,
 		Tags:     tags.Expand(t),
 		AzureFirewallPropertiesFormat: &network.AzureFirewallPropertiesFormat{
-			IPConfigurations: ipConfigs,
+			IPConfigurations:     ipConfigs,
+			AdditionalProperties: make(map[string]*string),
 		},
 		Zones: zones,
 	}
 
+	privateIpRanges := "IANAPrivateRanges"
+	if v, ok := d.GetOk("private_ip_ranges"); ok {
+		rangeSlice := utils.ExpandStringSlice(v.(*schema.Set).List())
+		if rangeSlice != nil {
+			privateIpRanges = strings.Join(*rangeSlice, ",")
+			parameters.AdditionalProperties["Network.SNAT.PrivateRanges"] = &privateIpRanges
```

Comment:
is there a reason these fields aren't exposed in the Swagger? historically the Networking API's have introduced breaking schema changes - so this is less likely if these are typed, since they'll be caught via ARM review?